### ISSUE TITLE
My question dump

### ISSUE THREAD

**Issue by MrMegarich** [rozniak/xfce-winxp-tc#234]
I got everything working fine, I just want to know how to compile the other themes (zune, metallic etc.)

**Comment by rozniak** [comment]
At the moment mate, Windows Classic, Professional/Watercolor, and Windows XP Style (Blue) are the only ones that get packaged. :open_mouth: 

The other themes atm only have XFWM4 borders done, I haven't gotten around to doing their GTK themes yet (been busy working on #205 lately).

**Comment by MrMegarich** [comment]
ok, but i have some more questions, how do i change the icons in the start menu? how do I rename stuff. Why does my winver have these ugly borders at the side?
![Uploading Screenshot_2023-11-18_17-19-58.png…]()

**Comment by rozniak** [comment]
Change the icons, or do you mean the programs that are listed?

Also your image doesn't seem to have uploaded!

**Comment by MrMegarich** [comment]
![Uploading Screenshot_2023-11-18_18-56-09.png…]()

**Comment by MrMegarich** [comment]
![Screenshot_2023-11-18_18-56-09](https://github.com/rozniak/xfce-winxp-tc/assets/95163334/22ef8872-1788-4b25-aa60-663b2d3ab228)

**Comment by MrMegarich** [comment]
Notice the ugly borders at the sides of the windows logo

**Comment by MrMegarich** [comment]
I'm at a family gathering sorry I couldn't get back to you sooner

**Comment by MrMegarich** [comment]
![Screenshot_2023-11-18_19-04-44](https://github.com/rozniak/xfce-winxp-tc/assets/95163334/500a4311-4806-4753-b19b-d78192e2d1aa)

**Comment by MrMegarich** [comment]
the my computer icon still has the linux mint logo !!!!!

**Comment by rozniak** [comment]
> ![Screenshot_2023-11-18_18-56-09](https://user-images.githubusercontent.com/95163334/283988945-22ef8872-1788-4b25-aa60-663b2d3ab228.png)

I figured this might be what you meant - it's a bug I need to sort out with `winver` when you have fonts larger than the original. I need to redo the whole dialog really because I made it when I knew far less about GTK.

> the my computer icon still has the linux mint logo !!!!!

This made me laugh, I've not seen that before. Perhaps somehow the vendor icon gets priority? I'll have to fire up Linux Mint again to investigate. :laughing:

**Comment by MrMegarich** [comment]
cheers

**Comment by MrMegarich** [comment]
also what is wintc-desktop?

**Comment by rozniak** [comment]
It is the program the spawns the desktop window. Though at the moment it simply paints with the default blue colour and has no functionality. The plan is to have it replace `xfdesktop4` eventually (so the desktop behaves the same as Windows XP).

**Comment by MrMegarich** [comment]
Are there any plans to add the windows xp shutdown menu?

**Comment by MrMegarich** [comment]
![image](https://github.com/rozniak/xfce-winxp-tc/assets/95163334/b2efceb6-8fea-4c3e-84be-4ae0f8f042eb)

**Comment by rozniak** [comment]
#178

**Comment by MrMegarich** [comment]
Thank you!

**Comment by MrMegarich** [comment]
Did you work out the my computer icon?

**Comment by rozniak** [comment]
No since I've been busy on #205 , I'll have to create a Linux Mint VM to investigate

**Comment by MrMegarich** [comment]
Ok thank you

**Comment by rozniak** [comment]
Sorry about the wait mate, I'm just setting up a Linux Mint VM to investigate atm

**Comment by rozniak** [comment]
@MrMegarich  Seems okay on mine, perhaps you have something under `~/.icons` that could cause it? If you had any other icon themes maybe? It should just use the `computer` icon, so it has to be being overridden somewhere.

![image](https://github.com/rozniak/xfce-winxp-tc/assets/13258281/ad12a6ce-6e30-4c78-a6b8-b09f14d5e4bc)

**Comment by MrMegarich** [comment]
strangest part is that I install the theme on 2 different Linux Mint themes and it did the same thing twice!

**Comment by MrMegarich** [comment]
I hate to bug you so much, but how do I install "liblightdm-gobject-dev" and "libwintc-msgina" ?

**Comment by rozniak** [comment]
I don't mind answering questions! :grin: 

The package names `./chkdeps.sh -l` gives you, you can bung straight into `apt`. So in this case you want `sudo apt install liblightdm-gobject-dev`

(the `-l` switch is to say 'only check for third-party dependencies', anything `*wintc*` comes from this repo as part of the build, so you're only interested in third-party libs)

`libwintc-msgina` is a new library used by `logonui`. I'd recommend doing another `./buildall.sh` and install everything outputted because there were some changes made to a few other libs too.

To use `logonui`, you will need to edit `/etc/lightdm/lightdm.conf` and set `greeter-session=wintc-logonui` and `sudo systemctl restart lightdm`

I would suggest testing in a VM first though!

**Comment by MrMegarich** [comment]
xfce-winxp-tc/packaging$ sudo apt install liblightdm-gobject-dev
[sudo] password for user:           
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package liblightdm-gobject-dev

I type sudo apt install but it just returns an error. please help

**Comment by rozniak** [comment]
Bollocks, I tested Arch and didn't test Ubuntu downstream... fail. Looks like the package name is different so I'll fix that.

**Comment by MrMegarich** [comment]
cheers

**Comment by rozniak** [comment]
Please pull latest `master`, it should provide you with the correct package name :sunglasses:

**Comment by MrMegarich** [comment]
Thanks dude!

**Comment by MrMegarich** [comment]
Another question: How to I get the custom bootvid?

**Comment by MrMegarich** [comment]
> All good, I actually worked it out between now and the last time we chatted, so pretty much, the computer.png icon somehow got replaced in icons/luna/devices with the mint logo and I had to manually go in and change it.

This is so annoying! It seems that whenever I restart the computer (fresh install) it automatically changes to the mint logo!

**Comment by rozniak** [comment]
> Another question: How to I get the custom bootvid?

It seems to be slightly different to what I'm used to - on Ubuntu and derivatives it seems to use the 'alternatives' system for setting the Plymouth theme:

```
sudo update-alternatives --install /usr/share/plymouth/default.plymouth default.plymouth 100 /usr/share/plymouth/bootvid/bootvid.plymouth
sudo update-alternatives --config default.plymouth
sudo update-initramfs -u
```

This should work - btw if you want to use a different SKU than 'Professional' you can pass the `-s` switch to `buildall.sh` to change it. The screenshot I took above was compiled using `./buildall.sh -s dnsrv-sbs` - it affects the banners and boot screen atm.

**Comment by rozniak** [comment]
I can confirm Linux Mint replaces all instances of `computer.png` on reboot under the theme, but I really haven't got a clue why it would be doing this
![image](https://github.com/rozniak/xfce-winxp-tc/assets/13258281/cc968527-fc06-4a2f-98c6-d04db7b1b5dd)

**Comment by MrMegarich** [comment]
> > Another question: How to I get the custom bootvid?
> 
> It seems to be slightly different to what I'm used to - on Ubuntu and derivatives it seems to use the 'alternatives' system for setting the Plymouth theme:
> 
> ```
> sudo update-alternatives --install /usr/share/plymouth/default.plymouth default.plymouth 100 /usr/share/plymouth/bootvid/bootvid.plymouth
> sudo update-alternatives --config default.plymouth
> sudo update-initramfs -u
> ```
> 
> This should work - btw if you want to use a different SKU than 'Professional' you can pass the `-s` switch to `buildall.sh` to change it. The screenshot I took above was compiled using `./buildall.sh -s dnsrv-sbs` . It affects the banners and boot screen atm.

I copy the commands you gave me into the terminal and it just asks me for an integer. Also thanks for the buildall tip, I was wondering how you got winver to say "Windows xp for small business"

**Comment by MrMegarich** [comment]
user@users-Nitro-AN515-56:~/Desktop$ sudo update-alternatives --install /usr/share/plymouth/default.plymouth default.plymouth 100 /usr/share/plymouth/bootvid/bootvid.plymouth
sudo update-alternatives --config default.plymouth
sudo update-initramfs -u
[sudo] password for user:           
update-alternatives: priority must be an integer

Use 'update-alternatives --help' for program usage information.
There are 2 choices for the alternative default.plymouth (providing /usr/share/plymouth/themes/default.plymouth).

  Selection    Path                                                     Priority   Status
------------------------------------------------------------
  0            /usr/share/plymouth/themes/mint-logo/mint-logo.plymouth   200       auto mode
  1            /usr/share/plymouth/themes/bgrt/bgrt.plymouth             110       manual mode
* 2            /usr/share/plymouth/themes/mint-logo/mint-logo.plymouth   200       manual mode

Press <enter> to keep the current choice[*], or type selection number:

**Comment by MrMegarich** [comment]
Copied straight from the terminal

**Comment by rozniak** [comment]
Edited to amend my error, rerun those commands - when it prompts you for a number, one of the available options should be `bootvid` :grin:

**Comment by MrMegarich** [comment]
I copy and pasted the commands again into the terminal, and now the bootvid is changed! ... to the ubuntu one

**Comment by MrMegarich** [comment]
at least it has my oem logo XD

**Comment by rozniak** [comment]
That's with bootvid showing as selected for default.plymouth and updated initramfs? It worked when I tested on Mint 🤔

**Comment by MrMegarich** [comment]
![Screenshot_2023-11-28_15-22-28](https://github.com/rozniak/xfce-winxp-tc/assets/95163334/5d90ba84-3bd0-43b4-95ec-a7bd3b3573da)

**Comment by MrMegarich** [comment]
> At the moment mate, Windows Classic, Professional/Watercolor, and Windows XP Style (Blue) are the only ones that get packaged. 😮
> 
> The other themes atm only have XFWM4 borders done, I haven't gotten around to doing their GTK themes yet (been busy working on #205 lately).

Hey seeing as how you've finished the greeter, does that mean that you now have time to work on the other themes? Or are you putting all your attention into the installer?

**Comment by rozniak** [comment]
The first command is wrong - the `100` needs to be the last argument -- notice it's complaining `priority must be an integer`

**Comment by MrMegarich** [comment]
So what would be the correct command?

**Comment by rozniak** [comment]
Move the `100` to be the last parameter, take another look at my comment earlier - I edited it to have the correct commands. :stuck_out_tongue:

**Comment by MrMegarich** [comment]
Damn Dag Nabbit! Now it's givin' me this error update-alternatives: error: alternative path /usr/share/plymouth/bootvid/bootvid.plymouth doesn't exist

**Comment by MrMegarich** [comment]
![Screenshot_2023-11-28_21-00-53](https://github.com/rozniak/xfce-winxp-tc/assets/95163334/b473f9e7-21e7-407c-b387-73c08efce612)

**Comment by rozniak** [comment]
I guess it is `/usr/share/plymouth/themes/bootvid/bootvid.plymouth`

**Comment by rozniak** [comment]
Always happy to help mate :smile: 

> Hey seeing as how you've finished the greeter, does that mean that you now have time to work on the other themes? Or are you putting all your attention into the installer?

Quite a few things need doing to be fair... the installer is currently a bit of a low priority item until there's enough completed to make the project worth having an installer for (and probably better to complete a few more packaging targets first).

I was planning to prioritise getting the volume icon working in the tray, look into some XFWM4 bugs/features upstream (because I could really do with them fixed upstream, I'm not forking again), and then probably some more programs/shell work.

**Comment by MrMegarich** [comment]
Awesome! The volume control and system tray will be really cool when it's finished.

**Comment by MrMegarich** [comment]
> I don't mind answering questions! 😁
> 
> The package names `./chkdeps.sh -l` gives you, you can bung straight into `apt`. So in this case you want `sudo apt install liblightdm-gobject-dev`
> 
> (the `-l` switch is to say 'only check for third-party dependencies', anything `*wintc*` comes from this repo as part of the build, so you're only interested in third-party libs)
> 
> `libwintc-msgina` is a new library used by `logonui`. I'd recommend doing another `./buildall.sh` and install everything outputted because there were some changes made to a few other libs too.
> 
> To use `logonui`, you will need to edit `/etc/lightdm/lightdm.conf` and set `greeter-session=wintc-logonui` and `sudo systemctl restart lightdm`
> 
> I would suggest testing in a VM first though!

I made the crucial mistake of running this on my host

**Comment by MrMegarich** [comment]
And after I ran sudo systemctl restart lightdm by pc wouldn't boot
it just got stuck on the bootvid.

**Comment by MrMegarich** [comment]
also do you know any way I can change the name of the Recycle Bin? (y'know so it says "Recycle Bin" and not Trash or Rubbish)

**Comment by rozniak** [comment]
> And after I ran sudo systemctl restart lightdm by pc wouldn't boot it just got stuck on the bootvid.

Sounds weird? It's permanently stuck like this ?

> also I don't have a file called lightdm.conf

I had a think, it might be another `update-alternatives` command that you need on Ubuntu/derivatives like how you changed the Plymouth boot screen. Should be able to search online for what you need to do to change the greeter on Ubuntu/Linux Mint.

> also do you know any way I can change the name of the Recycle Bin? (y'know so it says "Recycle Bin" and not Trash or Rubbish)

It's not really a standard thing afaik so there's no single place you can just bodge it. Usually these strings will be compiled in, and then if you're using a different locale other than `en-US` it will translate at runtime using `.po` files installed in `/usr/share/locale` . Essentially each program on Linux decides what to call it, so it could be "Rubbish Bin" in one spot, and "Trash" in another.

Eventually this won't really be much of a problem once more of the shell is done (eg. `wintc-desktop` and eventually an Explorer implementation). I already have all the standard place names defined in `libwintc-shllang` for this purpose: https://github.com/rozniak/xfce-winxp-tc/blob/master/shared/shllang/src/places.c#L20

**Comment by MrMegarich** [comment]
> > And after I ran sudo systemctl restart lightdm by pc wouldn't boot it just got stuck on the bootvid.
> 
> Sounds weird? It's permanently stuck like this ?
> 
> > also I don't have a file called lightdm.conf
> 
> I had a think, it might be another `update-alternatives` command that you need on Ubuntu/derivatives like how you changed the Plymouth boot screen. Should be able to search online for what you need to do to change the greeter on Ubuntu/Linux Mint.
> 
> > also do you know any way I can change the name of the Recycle Bin? (y'know so it says "Recycle Bin" and not Trash or Rubbish)
> 
> It's not really a standard thing afaik so there's no single place you can just bodge it. Usually these strings will be compiled in, and then if you're using a different locale other than `en-US` it will translate at runtime using `.po` files installed in `/usr/share/locale` . Essentially each program on Linux decides what to call it, so it could be "Rubbish Bin" in one spot, and "Trash" in another.
> 
> Eventually this won't really be much of a problem once more of the shell is done (eg. `wintc-desktop` and eventually an Explorer implementation). I already have all the standard place names defined in `libwintc-shllang` for this purpose: https://github.com/rozniak/xfce-winxp-tc/blob/master/shared/shllang/src/places.c#L20

Yeah it was permanently stuck like that, I had to boot of my usb and reinstall mint.

**Comment by MrMegarich** [comment]
Don't worry about the issues with the greeter I just got it set up

**Comment by MrMegarich** [comment]
How do I change the cursor on the greeter and how do I stop the windows flag from waving? [🤔]

**Comment by MrMegarich** [comment]
Righto. I found the solution to the greeter on the ubuntu wiki https://wiki.ubuntu.com/LightDM#Changing_the_Greeter. It was under the heading changing the greeter, I just copy pasted that into a lightdm.conf file and it worked.

**Comment by MrMegarich** [comment]
Will there be the bundled games in the future (e.g. minesweeper, solitaire etc) and will the Windows xp tour be included?

**Comment by rozniak** [comment]
> Will there be the bundled games in the future (e.g. minesweeper, solitaire etc) and will the Windows xp tour be included?

Yeah, from the project's README: `I don't have a fancy looking roadmap document for this repo - there's too much to list really. Essentially, if something was in Windows XP, it's on my mind.`

> also what distro would you recommend to run this project under?

Debian 12 would be the most 'tested' considering it's what I use daily, in theory anything with a supported package format in here should work though.

**Comment by MrMegarich** [comment]
Will you include a Rozniak Winxp-xfce Plus! package once this is finished?

**Comment by rozniak** [comment]
All the 'extras' could be considered wishlist items. Like anything from Plus!, Plus! DME, Tablet PC Fun Pack, Winter Packs etc. (it would be quite funny to 'port' over the desktop dancers for instance)

There'll probably also be a `reskit` for various tools and stuff (not replications of the Windows reskits -- more linux-focused)

**Comment by MrMegarich** [comment]
How do I get the custom sound theme? (I'm on mint xfce)

**Comment by MrMegarich** [comment]
Also, what are you currently working on?

**Comment by rozniak** [comment]
> How do I get the custom sound theme? (I'm on mint xfce)

You have to use the 'Settings Editor' under XFCE Settings - `xsettings > Net` and then set `SoundThemeName` to `Windows XP Default` and tick `EnableEventSounds` and `EnableInputFeedbackSounds`

> Also, what are you currently working on?

Are you watching the repo? I think GitHub would tell you if you were, otherwise you should be able to tell by looking at active branches:

![image](https://github.com/rozniak/xfce-winxp-tc/assets/13258281/e599a4d9-ff3b-4a8b-bc5f-937555277aaa)

Shows that I am currently working on #76

**Comment by MrMegarich** [comment]
Thanks for showing telling me about watching the repo, and for telling me about active branches (I hardly ever use github outside of some emulators and this comment thread).

**Comment by MrMegarich** [comment]
I hate to be nitpicky and I know that you're working on something else, but will there be the maximized toggled buttons (eventually, I know it's really really small but it would be so nice to have)

**Comment by rozniak** [comment]
I believe to do that would require it be implemented in XFWM4 upstream, depends whether they would accept such a feature or not.

**Comment by MrMegarich** [comment]
Well, what about linux mint-y? That theme has the Maximised button

**Comment by rozniak** [comment]
I just had a look, that does seem to be the case! `maximize-toggled-*` looks like what is needed, I didn't know that was a thing. :open_mouth: 

I will look into porting across the restore button glyphs after I sort out the changes I was working on.

**Comment by MrMegarich** [comment]
Hell yeah!

**Comment by rozniak** [comment]
Working on this in #254

**Comment by MrMegarich** [comment]
I'm glad I could bring that to your attention

**Comment by rozniak** [comment]
Pull latest `master` and recompile the themes, you should get the Restore caption button glyphs :sunglasses:

**Comment by MrMegarich** [comment]
Awesome

**Comment by MrMegarich** [comment]
That looks so much better, thanks alot!

**Comment by MrMegarich** [comment]
Also, are you considering a including an Internet Explorer 6 skin for firefox?

**Comment by MrMegarich** [comment]
Will there ever be the old Maximise, Minimise and Restore animations?

**Comment by MrMegarich** [comment]
And when notepad is done, will you add the "bush hid the facts" bug?

**Comment by rozniak** [comment]
> Will there ever be the old Maximise, Minimise and Restore animations?

It would mean forking XFWM4 to implement I think. If it ever happens it will be a long time before it does.

> And when notepad is done, will you add the "bush hid the facts" bug?

Probably not since I prefer to rely on existing libraries for things like text-encoding, It would seem like a lot of work just to replicate that bug otherwise.

**Comment by MrMegarich** [comment]
Will there be Control Panel, and all the little .cpl applets? Also, will there be the .msc programs from windows 2000? (I know, "If it's in xp it's on my mind", I just want to know for sure).

**Comment by rozniak** [comment]
Yes, the CPLs will be part of `explorer` though so I'll have to get all that working. As for the `msc` snap-ins I'll have to get an `mmc` recreation going also.

**Comment by MrMegarich** [comment]
What is shelldpa?

**Comment by rozniak** [comment]
https://github.com/rozniak/xfce-winxp-tc/blob/master/shared/shelldpa/README.MD

FYI every component's directory has a `README` that explains what it is - if you're interested in what each part of the project is/does, I encourage you to browse around. Should be `README`s for pretty much everything. :grin:

**Comment by MrMegarich** [comment]
oh thank you

**Comment by MrMegarich** [comment]
I have no clue what any of the stuff in the read me means 😅

**Comment by rozniak** [comment]
You know about X11 and Wayland?

Basically anything that needs to interact with the window manager (in this case, XFWM4) has to do so via either the X11 or Wayland display protocols. They're not compatible with each other, so the job of `libwintc-shelldpa` is to hide the difference between the two from programs.

For instance, `wintc-taskband` wants to anchor itself to one side of the primary display - X11 and Wayland both have their own ways of doing this. Rather than `wintc-taskband` having to deal with both itself, `libwintc-shelldpa` exposes `wintc_anchor_taskband_to_bottom()` which, depending on whether you're using X11 or Wayland, will perform the necessary operations to get the job done.

There are other APIs in there as well, eg. the taskbar buttons need to be able to control and listen to which window is currently active - `libwintc-shelldpa` hides the differences between X11 and Wayland for this also.

Of course, currently XFWM4 is X11 only. The Wayland support is there to future-proof this project for whenever the day comes that things switch over.

I hope this makes sense! :grin:

**Comment by MrMegarich** [comment]
Thanks for the explanation! Also, will there be taskbar locking and resizing, quick launch, and the option to switch to the Windows 9x style start menu (after you rewrite it of course).

**Comment by MrMegarich** [comment]
Also when you get around to the tour, I would recommend this website for the non-animated option! /[https://xptour.neocities.org/](url)

**Comment by MrMegarich** [comment]
I enabled event sounds and I did what you said to enable the sound theme but the only sound that is changed is the sound when I use the volume slider! Isn't there supposed to be a startup and shutdown sound? If so how do I enable them?

**Comment by rozniak** [comment]
> Also, will there be taskbar locking and resizing, quick launch, and the option to switch to the Windows 9x style start menu (after you rewrite it of course).

This is the kind of thing I mean by if Windows XP had it, it's something I will have thought about. :stuck_out_tongue: 

> Also when you get around to the tour, I would recommend this website for the non-animated option! 

This is just the same as the static HTML off the disc, no?

> I enabled event sounds and I did what you said to enable the sound theme but the only sound that is changed is the sound when I use the volume slider! Isn't there supposed to be a startup and shutdown sound? If so how do I enable them?

The sounds are indeed present, but they require programs to play them. XFCE itself doesn't use the sound theme at all from what I know. Sadly not many things actually do use the sound theme, some GTK or GNOME programs do (if you try and run something that doesn't exist in `run`, the GtkMessageDialog will play the critical stop sound).

Startup / shutdown sounds will be up to me to implement some program that plays them.

**Comment by MrMegarich** [comment]
Ah, yeah I guess I should have thought about all that

**Comment by MrMegarich** [comment]
Sorry

**Comment by rozniak** [comment]
No need to apologise mate! Always happy to help explain stuff if I can. :grin:

**Comment by MrMegarich** [comment]
Yo dude, I haven't posted here in a while and that's because I switched back to windows, but seeing microsofts new copilot button got me thinking about swtiching back to linux. If I do I will 100% be usuing this again to get that familiar windows feel (maybe once control panel and explorer are implemented)

**Comment by MrMegarich** [comment]
Also, I reckon you should make a discord server for this project

**Comment by rozniak** [comment]
No worries mate, I have been quite busy irl the past month or so I'm just trying to find time and tune back into the project again atm.

**Comment by MrMegarich** [comment]
All good.
